packages/expo-modules-core/android/src/main/java/expo/modules/kotlin/KotlinInteropModuleRegistry.kt

```diff
@@ -41,26 +41,22 @@ class KotlinInteropModuleRegistry(
   }
 
   fun exportedModulesConstants(): Map<ModuleName, ModuleConstants> {
-    return registry
-      .map { holder ->
-        holder.name to holder.definition.constantsProvider()
-      }
-      .toMap()
+    return registry.associate { holder ->
+      holder.name to holder.definition.constantsProvider()
+    }
   }
 
   fun exportMethods(exportKey: (String, List<ModuleMethodInfo>) -> Unit = { _, _ -> }): Map<ModuleName, List<ModuleMethodInfo>> {
-    return registry
-      .map { holder ->
-        val methodsInfo = holder.definition.methods.map { (name, method) ->
-          mapOf(
-            "name" to name,
-            "argumentsCount" to method.argsCount
-          )
-        }
-        exportKey(holder.name, methodsInfo)
-        holder.name to methodsInfo
+    return registry.associate { holder ->
+      val methodsInfo = holder.definition.methods.filter { (_, method) -> method.isSync }.map { (name, method) ->
```

Comment:
- Why do you filter here only sync methods? They weren't filtered before so the asyncs were exported here
- I'd split this, IMO it's difficult to read now:
   ```suggestion
         val methodsInfo = holder.definition.methods
           .filter { (_, method) -> method.isSync }
           .map { (name, method) ->
   ```

Reply:
It was a mistake - here we want to export only async functions.